pgtap/topology/createMultimodalGraph/2-lines-0 points-0 z-0 .sql

```diff
@@ -0,0 +1,206 @@
+\i setup.sql
+
+set client_min_messages TO warning;
+
+select plan(13);
+drop TABLE IF EXISTS test_table_l1;
+create table test_table_l1(
+   geom geometry('linestringz',4326),
+   id integer primary key
+);
+--1rst group
+insert into test_table_l1 VALUES ('SRID=4326;linestring(5 0 0,10 10 0, 13 10 0, 15 10 0)',1);
+insert into test_table_l1 VALUES ('SRID=4326;linestring(0 0 0, 10 10 0)',2);
+insert into test_table_l1 VALUES ('SRID=4326;linestring(10 10 0, 10 0 0)',3);
+insert into test_table_l1 VALUES ('SRID=4326;linestring(8 0 0, 10 10 0)',4);
+insert into test_table_l1 VALUES ('SRID=4326;linestring(8 0 0, 8 10 0, 10 10 0)', 5);
+insert into test_table_l1 values ('SRID=4326;linestring(7 12 0, 13 10 0, 14 8 0)', 6);
+
+--for test z -2nd group
+insert into test_table_l1 values ('SRID=4326;linestring(15 14 50, 15 10 50, 15 8 50)', 7); --overpass over point(15 10 0)
+insert into test_table_l1 values ('SRID=4326;linestring(15 16 35, 15 14 50)', 8); -- z connects with edge points
+insert into test_table_l1 values ('SRID=4326;linestring(13 16 35, 15 14 50, 13 14 50)', 9); -- z connects with interior points
+
+
+--2nd layer
+drop TABLE IF EXISTS test_table_l2;
+create table test_table_l2(
+  geom geometry('linestringz',4326),
+  id integer primary key
+);
+
+insert into test_table_l2 values ('SRID=4326;linestring(13 18 35, 13 16 35, 7 12 0)', 1);
+insert into test_table_l2 values ('SRID=4326;linestring(15 18 0, 15 16 35, 17 18 0)', 2);
+
```

Comment:
Yes it can be done. The only change this will bring on is when calling the algorithm.
Each table is a representation of a set of edges. If two tables are joined like you mentioned and you want to split set of edges when building the graph, you will have to select from same table but split data using a where.

Instead of :
```
prepare createTopology_1 as
  SELECT count(*) from  pgr_create_multimodal_graph( 
... , ... ,
'{
  "linealLayer-1": {
    "sql": "select id as id, geom as the_geom,0 as z_start, 0 as z_end 
              from \"test_table_l1\"",
    ...
  },"linealLayer-2": {
    "sql": "select id as id, geom as the_geom,0 as z_start, 0 as z_end 
             from \"test_table_l2\"",
    ...
  },
  ...
}', 'graph_lines', 'public', 0.000001);
```

you will have to do;

```
prepare createTopology_1 as
  SELECT count(*) from  pgr_create_multimodal_graph( 
... , ... ,
'{
  "linealLayer-1": {
    "sql": "select id as id, geom as the_geom,0 as z_start, 0 as z_end 
              from \"test_table_l1\"
              where id < n",
    ...
  },"linealLayer-2": {
    "sql": "select id as id, geom as the_geom,0 as z_start, 0 as z_end 
             from \"test_table_l1\"
             where id >= n",
    ...
  },
  ...
}', 'graph_lines', 'public', 0.000001);
```